### PR TITLE
Add Status Update Functionality and "Clear All Completed" Button

### PR DESCRIPTION
**Pull request by ClarenceAtim**
**This pull request adds new functionality to the existing codebase, enabling users to update the status of tasks and clear all completed tasks.**

Changes Made:

- Added a function to clear all completed tasks using the filter method.
- Updated the index of remaining tasks after clearing completed tasks.
- Enhanced the UI by refreshing the task list after clearing completed tasks.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import './style.css';
 import Data from './modules/data.js';
-import { getItems, addToLocalStorage } from './modules/utility.js';
+import { getItems, addToLocalStorage, clearCompletedTasks } from './modules/utility.js';
```
Comment (codepantha) [inline src/index.js:3]: - [ ] After marking a task as completed, it is updated in the database 👏 However, if the page is reloaded, the checkbox of the task that is been marked as completed is no longer checked ☑️ even though `completed === true` in localstorage. Please fix this as it is very important for user experience.
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ClarenceAtim 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Tasks can be marked as completed 📌 
✔️ Updated tasks are stored in localStorage 📌 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ClarenceAtim** [comment]
Updated the displayItems method in the UI class in the ui.js module.

checked
property of the checkbox based on the
completed
property of the item when the page is loaded.

**Review by muneebulrehman — APPROVED** [review]
Hi @ClarenceAtim , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- All functionality working as expected
- Great follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Hello @ClarenceAtim 👋,

_This is an additional review as a result of the second opinion request._

I can see that your project has been approved.

PS: Next time you request a Second Opinion Review, please wait for it before asking for a regular review again. While it may take longer to receive the second opinion, it will save a review for you in case there are any issues. Additionally, please refrain from pushing any commits after receiving the initial review so that we can examine the project again and provide a comprehensive second opinion.

Keep rocking 🚀!

**Pull request merged**